src/zimscraperlib/zim/creator.py

```diff
@@ -146,7 +150,35 @@ def config_indexing(
         self.__indexing_configured = True
         return self
 
+    def _is_illustration_metadata_name(self, name: str) -> bool:
+        """Return True if name is a valid illustration metadata name"""
+        return name.startswith("Illustration_")
+
+    def _get_illustration_metadata_details(
+        self,
+        value: bytes,
+    ) -> str | None:
+        """Return image format for debug logging of illustration metadata"""
+        try:
+            with PIL.Image.open(io.BytesIO(value)) as img:
+                if img is not None:
+                    return f"{img.format} {img.size[0]}x{img.size[1]}"
+        except PIL.UnidentifiedImageError as e:
+            return f"Image format issue: {e}"
+        return f"Unknown image format, {len(value)} bytes"
+
+    def _log_metadata(self):
+        for name, value in sorted(self._metadata.items()):
+            if self._is_illustration_metadata_name(name) and isinstance(value, bytes):
+                illus_md = self._get_illustration_metadata_details(value)
+            else:
```

Comment:
This will log everything that's not `bytes`. What if it cannot be represented as string?

Reply:
That would indeed be unfortunate. The preferred ways to set up metadata, though, go through typed handlers that ensure it will be bytes. Worst case, IIUC, is that it gets logged as is. That might be annoying if it's a 47K character string or somesuch, but it's not terrible.